### PR TITLE
Change topic selector

### PR DESCRIPTION
**Pull request by loicgasser**
[Test Link](https://mf-geoadmin3.int.bgdi.ch/gal_topic_selector/index.html)

Relates to discussions in:
https://github.com/geoadmin/mf-geoadmin3/issues/3157
https://github.com/geoadmin/mf-geoadmin3/pull/3592

It would be good to test on all platform before merging this.

### CODE REVIEW

**Comment by davidoesch** [comment]
@mariokeusen can you test on dekstop plattforms? I'll test mobile

feedback so far

- [ ] "thema wechseln" as button is ugly, I would rather stick to text and icon as it is now
- [ ] scroll the modal window rather than in the app

image

**Comment by loicgasser** [comment]
This is not the latest version something went wrong with the deploybranch sorry

**Comment by loicgasser** [comment]
image

Ok so leave button as it is now in geoadmin.
Tune modal max max-height a bit.

File: src/index.mako.html [hunk 1]
```diff
@@ -420,17 +420,9 @@
...
-            <a class="dropdown-toggle" data-toggle="dropdown" href="#">
-              <span translate>choose_theme</span>
-              <i class="fa fa-chevron-sign-right"></i>
-            </a>
-            <div class="dropdown-menu">
-              <div ga-topic ga-topic-ui="list"></div>
-            </div>
-          </div>
-  % endif
+          <button type="button" class="btn btn-default pull-right topic-btn" data-toggle="modal" data-target=".ga-topic-modal">
```
Comment (oterral) [inline src/index.mako.html]: can you use an identifier for the data-target like this `data-target="#topicSelector .modal"` ?   A class could potentially be not unique.

File: src/components/topic/partials/topic.html [hunk 2]
```diff
@@ -1,8 +1,22 @@
-<div class="ga-topic-item"
-     ng-repeat="topic in ::topics"
-     ng-click="setActiveTopic(topic)"
-     ng-class="{'ga-topic-active': topic == activeTopic}"
-     data-original-title="{{topic.tooltip | translate}}">
-  <div translate>{{::topic.id}}</div>
-  <div class="ga-topics-sprite ga-topics-sprite-{{::topic.id}}"></div>
+<div class="modal fade ga-topic-modal">
```
Comment (oterral) [inline src/components/topic/partials/topic.html]: no need to use a `ga-topic-modal` class .

File: src/components/topic/TopicDirective.js [hunk 3]
```diff
@@ -11,12 +11,11 @@ goog.require('ga_topic_service');
...
           restrict: 'A',
-          templateUrl: function(element, attrs) {
-            return 'components/topic/partials/topic.' +
-              ((attrs.gaTopicUi == 'select') ? 'select.html' : 'html');
-          },
+          templateUrl: 'components/topic/partials/topic.html',
           scope: {},
           link: function(scope, element, attrs) {
+            var modalClsName = 'ga-topic-modal';
+            var modal = element.find('div.' + modalClsName);
```
Comment (oterral) [inline src/components/topic/TopicDirective.js]: Use the `.modal` class instead`.   More generic and we don't need a constant to get the correct element.

File: src/components/topic/TopicDirective.js [hunk 4]
```diff
@@ -37,6 +37,7 @@ goog.require('ga_topic_service');
               scope.activeTopic = gaTopic.get();
               scope.$applyAsync(function() {
                 element.find('.ga-topic-item').tooltip({
+                  container: 'div.' + modalClsName,
```
Comment (oterral) [inline src/components/topic/TopicDirective.js]: Maybe we could remove all these toolitps and use the tilte attribute ? what do you think?

**Comment by oterral** [comment]
I've tested a bit and it seems on tablet there is css adjusments to make.

**Comment by davidoesch** [comment]
On mobile, modal can be have a larger height
screenshot_20161115-160703

**Comment by loicgasser** [comment]
Updated css were stuck in varnish cache.
Please test review again

**Comment by mariokeusen** [comment]
Made a few tests on desktop (FF, IE, Chrome): it works and I like it!

File: src/components/topic/partials/topic.html [hunk 5]
```diff
@@ -1,8 +1,22 @@
-<div class="ga-topic-item"
-     ng-repeat="topic in ::topics"
-     ng-click="setActiveTopic(topic)"
-     ng-class="{'ga-topic-active': topic == activeTopic}"
-     data-original-title="{{topic.tooltip | translate}}">
-  <div translate>{{::topic.id}}</div>
-  <div class="ga-topics-sprite ga-topics-sprite-{{::topic.id}}"></div>
+<div id="topicSelector" class="modal fade">
```
Comment (oterral) [inline src/components/topic/partials/topic.html]: no id in hte partial . I meant putting it in the index.mako.html
```
<div id="topicSelector" ga-topic>
```

File: src/components/topic/TopicDirective.js [hunk 6]
```diff
@@ -37,6 +37,7 @@ goog.require('ga_topic_service');
               scope.activeTopic = gaTopic.get();
               scope.$applyAsync(function() {
                 element.find('.ga-topic-item').tooltip({
+                  container: modalId,
```
Comment (oterral) [inline src/components/topic/TopicDirective.js]: can't we avoid the use of container?

**Comment by davidoesch** [comment]
Nice. tested on different Brwosers and devices. especially on mobile tablets it way more useful than the wheel

one thing to fix: no button only text (as the delete and share button in draw) and remove the icon (we don't use icons in the UI--KISS)

image

**Comment by loicgasser** [comment]
KISS: a button is simple, a hidden anchor is not

Let me try to convice you a bit here. **I can ensure you nobody will see it**, even less than with the current design. Maybe you don't like the design in this PR, I guess it's a matter of taste, but we should really try to make an effort to show this feature to our users. Make sure people understand they can click it and change topic. We alwas say that people don't change topic, maybe there is a reason. During usability tests people never realized they were able to change the topic. Don't you have another idea?

**Comment by davidoesch** [comment]
A button won't solve the issue that ppl don't find the topics. this has to be solved with more than a button
it is about consistency in the UI: we simply don't have buttons in the accordion top level. We don't use icons in the UI of map.geo.admin.ch. WHY? because we introduced removed and reintroduced and removed icons  and the background is quite well described here http://mapbrief.com/2013/02/19/an-iconography-of-confusion-why-map-portals-dont-work-part-iv/

we will change the access to  topics, completely 
there appraoches like this
image
or even in an ideal world will have SEO whcih will guide users directly to topics or have topics in the search as well... so with UI now we just do patchwork

Conclusio: we stick to our CI / CD . and when we fix it, we fix it completely. we have a link currently and use now a link as well. this  issue was about fixing the height of the topic selector. we fix that first and then we move/improve topics.

**Comment by loicgasser** [comment]
ok no problem

**Comment by loicgasser** [comment]
You notice the use of the same icon though. ;)

File: src/components/topic/TopicDirective.js (hunk 6 above)
Comment (loicgasser) [inline src/components/topic/TopicDirective.js]: If not defined tooltip on topics pushes the modal content aside.

**Comment by loicgasser** [comment]
comments addressed

**Comment by oterral** [comment]
@davidoesch are you happy with the design

**Comment by davidoesch** [comment]
+1 clear go

**Pull request merged**